src/plugins/data/public/search/session/session_service.ts

```diff
@@ -128,6 +128,21 @@ export class SessionService {
       this.subscription.add(
         coreStart.application.currentAppId$.subscribe((newAppName) => {
           this.currentApp = newAppName;
+          if (!this.getSessionId()) return;
+
+          // Apps required to clean up their sessions before unmounting
```

Comment:
Is this change expected or is this a bad merge? Wasn't this removed with? https://github.com/elastic/kibana/pull/100214. FYI, on a related note: https://github.com/elastic/kibana/issues/101482